client/src/main/java/io/confluent/kafka/schemaregistry/avro/AvroSchemaUtils.java

```diff
@@ -226,10 +267,11 @@ public static byte[] toJson(Object value) throws IOException {
     }
   }
 
+  @SuppressWarnings("unchecked")
   public static void toJson(Object value, OutputStream out) throws IOException {
     Schema schema = getSchema(value);
     JsonEncoder encoder = encoderFactory.jsonEncoder(schema, out);
-    DatumWriter<Object> writer = new GenericDatumWriter<>(schema);
+    DatumWriter<Object> writer = (DatumWriter<Object>) getDatumWriter(value, schema, true);
```

Comment:
Is this a bug that previously this method always use a GenericDatumWriter? We don't have a test for converting a SpecificRecord to Json so not sure how the behavior will change there.

Reply:
It wasn't a bug, since it was always being passed a GenericRecord.  But for some upcoming code I want to pass other types of objects.